### PR TITLE
Add MapView component

### PR DESCRIPTION
**Pull request by sayas01**
This PR includes:
1) Addition of a mapbox gl map component with markers for db entries based on the zoom level.
2) A pop up on feature click with the feature details. The styling of the tooltip still requires some work.

DNM status:
The recent updates to the MapView component could not be tested because of the invocation limit issue with Firebase. Should be merged only after testing has been completed.

### CODE REVIEW

File: functions/index.js [hunk 1]
```diff
@@ -69,30 +69,28 @@ exports.getReports = functions.https.onRequest((req, res) => {
...
-                // If distance is within 1 mile from the query lat long
-                if (distance <= 1) {
-                  items.push({ id: doc.id, data: doc.data() });
-                }
+            const dataLatitude = data['mapLat'];
+            const dataLongitude = data['mapLng'];
+            if (dataLatitude !== undefined && dataLongitude !== undefined) {
+              const to = turf.point([dataLongitude, dataLatitude]);
+              const distance = turf.distance(from, to, options);
+              console.log(distance)
```
Comment (savannahostrowski) [inline functions/index.js]: Let's remove this.

Comment (savannahostrowski) [inline functions/index.js]: Nit: remove this console.log

File: functions/index.js [hunk 2]
```diff
@@ -69,30 +69,28 @@ exports.getReports = functions.https.onRequest((req, res) => {
...
+              // If distance is within 1 mile from the query lat long
+              if (distance <= 1000) {
```
Comment (savannahostrowski) [inline functions/index.js]: Let's reduce this a little and update the comment above.

File: functions/index.js [hunk 3]
```diff
@@ -69,30 +69,28 @@ exports.getReports = functions.https.onRequest((req, res) => {
...
+                items.push({ id: doc.id, data: doc.data() });
               }
             }
           });
         } else {
           snapshot.forEach(doc => {
+            console.log(doc)
```
Comment (savannahostrowski) [inline functions/index.js]: Nit: remove.

File: src/components/MapView.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    };
+
+    getColor(species) {
+        return speciesColorMap.get(species)!== undefined ? speciesColorMap.get(species) : '#805b14';
+    }
+
+    renderPopup() {
+        const {popupInfo} = this.state;
+
+        return popupInfo && (
```
Comment (savannahostrowski) [inline src/components/MapView.js]: Mmm is this `popupInfo &&` checking that the popupInfo is defined before we try and render the content?

File: src/components/MapView.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        }
+        return (
+            <div className="mapContainer ">
+                <Map2 style="mapbox://styles/mapbox/streets-v9"
+                      className="map"
+                      {...this.state.viewport}
+                      onMoveEnd={e => this.onMoveEnd(e)}
+                      >
+                    {this.renderPopup()}
+                    {reports.map((report) => (
```
Comment (savannahostrowski) [inline src/components/MapView.js]: Nit: don't need brackets around one arg functions

File: src/components/PointTooltip.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+import CardMedia from '@material-ui/core/CardMedia';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+
+const DEFAULT_IMAGE_URL = "https://upload.wikimedia.org/wikipedia/commons/thumb/7/79/2010-brown-bear.jpg/200px-2010-brown-bear.jpg";
+
+const timeToString = time => {
```
Comment (savannahostrowski) [inline src/components/PointTooltip.js]: Mm, we have a similar function in the ListCard component I think (might be in my open PR). Once merged, maybe we can pull this function into a util.js file so we can use it globally?

File: src/components/PointTooltip.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+};
+
+const PointTooltip = props => {
+    const { data } = props;
+    return <div width = {100}>
+        <img width={50} src={data.image_url ? data.image_url : DEFAULT_IMAGE_URL} />
+        <div>{data.species}</div>
+        <div>Date & Time:</div>
+        <div>{timeToString(data.timestamp)}</div>
+        <div>Location: {data.mapLat},{data.mapLng}</div>
```
Comment (savannahostrowski) [inline src/components/PointTooltip.js]: We probably won't want this but rather just the neighbourhood

File: src/components/PointTooltip.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    </div>
+};
+
+export default PointTooltip;
```
Comment (savannahostrowski) [inline src/components/PointTooltip.js]: Nit new line!

File: src/components/PointTooltip.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+const timeToString = time => {
+    const nanosInMillis = time._nanoseconds / 1000000;
+    const secondsInMillis = time._seconds * 1000;
+    return new Date(nanosInMillis + secondsInMillis).toLocaleString();
+};
+
+const PointTooltip = props => {
+    const { data } = props;
+    return <div width = {100}>
```
Comment (savannahostrowski) [inline src/components/PointTooltip.js]: Nit: spaces on either side of the =

File: functions/index.js (hunk 2 above)
Comment (sayas01) [reply functions/index.js]: What about 500 mile radius?

File: src/components/MapView.js (hunk 4 above)
Comment (sayas01) [reply src/components/MapView.js]: Yes! I just want to render it if we have the content. Do you think there is a better way to do it?

File: src/components/PointTooltip.js (hunk 7 above)
Comment (sayas01) [reply src/components/PointTooltip.js]: yeah, I just added the location until we have the neighborhood. Will add a comment before this line to keep track of it.

File: functions/index.js (hunk 2 above)
Comment (savannahostrowski) [reply functions/index.js]: Sure - sounds good.

File: src/components/MapView.js (hunk 4 above)
Comment (savannahostrowski) [reply src/components/MapView.js]: Yeah! So I typically use something like this:

` if (!reports) {
      return <CircularProgress/>;
    }`

This checks to see if the reports are defined. If not, we just render a circular spinner. Once the content populates, the component automatically re-renders and we can render the actual content.

File: src/components/PointTooltip.js (hunk 7 above)
Comment (savannahostrowski) [reply src/components/PointTooltip.js]: Sounds good!

File: src/components/MapView.js (hunk 4 above)
Comment (sayas01) [reply src/components/MapView.js]: will update it! Thank you!

File: src/components/MapView.js [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+                    this.setState({reports: null})
+            })
+            .catch(error => error);
+
+    };
+
+    getColor(species) {
+        return speciesColorMap.get(species)!== undefined ? speciesColorMap.get(species) : '#805b14';
+    }
+    // getReport(report) {
```
Comment (savannahostrowski) [inline src/components/MapView.js]: Nit: let's remove this

File: src/components/MapView.js [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+        axios.get(url)
+            .then(reports => {
+                reports.data!=="No data!" ? this.setState({ reports: reports.data}):
+                    this.setState({reports: null})
+            })
+            .catch(error => error);
+
+    };
+
+    getColor(species) {
```
Comment (savannahostrowski) [inline src/components/MapView.js]: This should be an arrow function

`getColor = species => {..}`

File: src/components/MapView.js [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+        return speciesColorMap.get(species)!== undefined ? speciesColorMap.get(species) : '#805b14';
```
Comment (savannahostrowski) [inline src/components/MapView.js]: Could this just be `!speciesColorMap.get(species) ? ...`

File: src/components/MapView.js [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+    renderPopup() {
+        const {popupInfo} = this.state;
+        if(popupInfo)
+        {
+            return (
+                <Popup tipSize={5}
+                       anchor="bottom"
+                       coordinates={[popupInfo.mapLng, popupInfo.mapLat]}
+                       className="cardContainer"
+                    //onClick={() => this.getReport(popupInfo)}
```
Comment (savannahostrowski) [inline src/components/MapView.js]: Is this intentional commented out?

Comment (sayas01) [reply src/components/MapView.js]: Yes, I wanted to discuss with you about the view report on popup click before implementing it. That is why I commented this and getReport().

Comment (savannahostrowski) [reply src/components/MapView.js]: Ah okay - we should probably have that discussion before this gets merged.

**Pull request merged**